kotlin/feature-layer-query/src/main/java/com/esri/arcgisruntime/sample/featurelayerquery/MainActivity.kt

```diff
@@ -0,0 +1,150 @@
+import com.esri.arcgisruntime.geometry.Point
+import com.esri.arcgisruntime.geometry.SpatialReferences
+import com.esri.arcgisruntime.layers.FeatureLayer
+import com.esri.arcgisruntime.mapping.ArcGISMap
+import com.esri.arcgisruntime.mapping.Basemap
+import kotlinx.android.synthetic.main.activity_main.*
+import java.lang.Exception
+import java.util.Locale
+
+class MainActivity : AppCompatActivity() {
+
+  private val serviceFeatureTable: ServiceFeatureTable by lazy {
+    ServiceFeatureTable(getString(R.string.us_daytime_population_url))
+  }
+
+  private val featureLayer: FeatureLayer by lazy {
+    FeatureLayer(serviceFeatureTable)
+  }
+
+  override fun onCreate(savedInstanceState: Bundle?) {
+    super.onCreate(savedInstanceState)
+    setContentView(R.layout.activity_main)
+
+    featureLayer.apply {
+      opacity = 0.8f
+      maxScale = 10000.0
+    }
+
+    mapView.apply {
+      map = ArcGISMap(Basemap.createTopographic())
+    }.run {
+      map.operationalLayers.add(featureLayer)
+      setViewpointCenterAsync(
+        Point(-11000000.0, 5000000.0, SpatialReferences.getWebMercator()),
+        100000000.0
+      )
+    }
+
+  }
+
+  /**
+   * Handle the search intent from the search widget
+   */
+  override fun onNewIntent(intent: Intent) {
+    setIntent(intent)
+    if (Intent.ACTION_SEARCH == intent.action) {
+      val searchString: String? = intent.getStringExtra(SearchManager.QUERY)
+      if (searchString != null && !searchString.isEmpty())
+        searchForState(searchString)
+    }
+  }
+
+  private fun searchForState(searchString: String) {
+    // clear any previous selections
+    featureLayer.clearSelection()
+    // create objects required to do a selection with a query
+    val query = QueryParameters()
+    // make search case insensitive
+    query.whereClause =
+      ("upper(STATE_NAME) LIKE '%" + searchString.toUpperCase(Locale.US) + "%'")
+    // call select features
+    val future: ListenableFuture<FeatureQueryResult> = serviceFeatureTable.queryFeaturesAsync(query)
+    // add done loading listener to fire when the selection returns
+    future.addDoneListener {
+      try {
+        // call get on the future to get the result
+        val result = future.get()
+        // check there are some results
+        val resultIterator = result.iterator()
+        if (resultIterator.hasNext()) {
+          resultIterator.next().run {
+            // get the extent of the first feature in the result to zoom to
+            val envelope = geometry.extent
+            mapView.setViewpointGeometryAsync(envelope, 10.0)
+            // select the feature
+            featureLayer.selectFeature(this)
+          }
+        } else {
+          Toast.makeText(this, "No states found with name: $searchString", Toast.LENGTH_LONG).show()
+        }
+      } catch (e: Exception) {
+        val error = "Feature search failed for: $searchString. Error: ${e.message}"
+        Toast.makeText(this, error, Toast.LENGTH_LONG).show()
```

Comment:
Let's write out to Log.e too in case someone is working through log cat too

Reply:
alright, it seems readable and appropriate to do something like this 
`
"Feature search failed for: $searchString. Error: ${e.message}".also {
  Log.e(TAG, it)
  Toast.makeText(this, it, Toast.LENGTH_LONG).show()
}   
`

Reply:
Really like it 👍